File: Mixter.Domain/Core/Messages/TimelineMessageId.cs
using Mixter.Domain.Identity;

namespace Mixter.Domain.Core.Messages
{
    public struct TimelineMessageId
    {
        public UserId Owner { get; private set; }

        public MessageId MessageId { get; private set; }

        public TimelineMessageId(UserId owner, MessageId messageId)
            : this()
        {
            Owner = owner;
            MessageId = messageId;
        }
    }
}

File: app/Domain/Timeline/TimelineMessageId.php
<?php

namespace App\Domain\Timeline;

use App\Domain\Identity\UserId;
use App\Domain\Messages\MessageId;

class TimelineMessageId
{
    /**
     * @var MessageId
     */
    private $messageId;

    /**
     * @var UserId
     */
    private $ownerId;

    public function __construct(MessageId $messageId, UserId $ownerId)
    {
        $this->messageId = $messageId;
        $this->ownerId = $ownerId;
    }

    public function getId()
    {
        return $this->messageId->getId().'_'.$this->ownerId->getId();
    }
}